### DISCUSSION TITLE
CCXT Pair Converter Behavior Change & Clarification on Alias Updates

### DISCUSSION THREAD

**Discussion by ndopencode** [eprbell/dali-rp2#206] [Q&A]
I've observed some behavior changes in the recent alias updates to the CCXT Pair Converter. Previously, when I reviewed my transactions, I noticed that the spot prices were derived with a granularity of 60 minutes, presumably using the Kraken CSV pair converter. However, with the current version of dali-rp2, the granularity seems to have been reduced to 1 week, which has led to differences in the spot prices for my transactions.

Upon further investigation, the change in behavior seems to be linked to the following line of code in ccxt.py:

**ccxt.py:783**
```
bar_check = self.find_historical_bars(child_name, neighbor.name, week_start_date, self.__exchange_markets[exchange][child_name + neighbor.name][0], True, _ONE_WEEK)
```
Given this observation, I have a few specific inquiries:

**Default Behavior**: Is it possible to revert the CCXT Pair Converter's behavior to match the previous Kraken CSV pair converter? I value the ability to understand the previous implementation before transitioning to a new one, as it aids my comprehension and review of updates.

**Function Clarification**: Could someone provide explanations for the following functions in ccxt.py?
- _cache_graph_snapshots at line 634
- _optimize_assets_for_exchange at line 742

**Alias Update Overview**: I'd appreciate an overview of the recent alias updates. Specifically, I'm interested in understanding both the no-cache and cached operational flows that the pair converter utilizes to retrieve spot prices. This knowledge is crucial for me, especially in terms of its impact on my tax calculations in rp2.

Thank you to those who helped get this update through. I look forward to understanding these changes and ensuring that my calculations remain accurate.

**Comment by eprbell** [comment]
This is a question for @macanudo527.

**Comment by macanudo527** [comment]
I'll have to see if this is a bug of some kind. Line 783 is used for optimization and not for pricing so that shouldn't be affecting the pricing.

I think it is best to open this as an issue focused on incorrect pricing and then we can go from there.

**Comment by macanudo527** [comment]
The pricing issue should be resolved with #207 . Let me know if it doesn't.
